### PR TITLE
pkg/trace/obfuscate: remove obfuscations count metric

### LINKED ISSUE #6288: Move or remove datadog.trace_agent.obfuscations metric
In some trace-agent profiles, the SQL obfuscator spends a huge amount of time in `obfuscate.(*Obfuscator).obfuscateSQL.func1` doing a `metrics.Count`.

I believe that the function and the `metrics.Count` is in this defer here: https://github.com/DataDog/datadog-agent/blob/c43034dc8b892fafd22421b19d07bb54aa0d6b67/pkg/trace/obfuscate/sql.go#L311-L313

We should reproduce and confirm that this is the issue and move or remove the counter elsewhere to keep it off the hot path.

### PR DESCRIPTION
**Pull request by knusbaum**
The datadog.trace_agent.obfuscations count is on the obfuscation hot path,
and has been found to cause slowdowns on occasion. This metric is not very
useful and so is being removed in this commit.

Fixes #6288

### Motivation

We have seen trace-agent profiles indicating that this metric can cause slowdowns

### CODE REVIEW

File: releasenotes/notes/remove-obfuscations-metric-c082f6fd63085c67.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    APM: Remove the datadog.trace_agent.obfuscations count which can slow
```
Comment (gbbr) [inline releasenotes/notes/remove-obfuscations-metric-c082f6fd63085c67.yaml:11]: We don't need a change log on this PR. Just add the `no-changelog` label. This isn't something users care about or use. If it were, it'd have been more of a challenge to remove it.

**Review by gbbr — APPROVED** [review]

File: pkg/trace/obfuscate/sql.go [hunk 2]
```diff
@@ -326,18 +320,15 @@ func (o *Obfuscator) obfuscateSQL(span *pb.Span) {
...
 	}
 
-	tags = append(tags, "outcome:success")
 	span.Resource = oq.Query
 
 	if len(oq.TablesCSV) > 0 {
 		traceutil.SetMeta(span, "sql.tables", oq.TablesCSV)
 	}
 	if span.Meta != nil && span.Meta[sqlQueryTag] != "" {
-		// "sql.query" tag already set by user, do not change it.
```
Comment (gbbr) [inline pkg/trace/obfuscate/sql.go:340]: Why did you remove this? We need this comment. When the `sql.query` tag is set by the user we don't override it.

**Review by gbbr — CHANGES_REQUESTED** [review]

File: pkg/trace/obfuscate/sql.go (hunk 2 above)
Comment (knusbaum) [reply pkg/trace/obfuscate/sql.go:340]: Sorry, I misread this comment.
I will put it back.
